### ISSUE TITLE
Deaf quirk makes you permanently deaf, even if you lose it.

### ISSUE THREAD

**Issue by Onule** [tgstation/tgstation#60809]
## Testmerges:

#60317: 'large refactor of machine/power code to cut down on processing time and wasted lists' by Kylerace at commit 7e4829934d
#60771: 'Added component manipulation on objects' by Watermelon914 at commit fcdde790ac
#60755: 'Changes the input list in VV add component to tgui_input_list()' by Thunder12345 at commit cf62458dfa

## Reproduction:

Getting the Deaf quirk via Eigenstasium OD will make you permanently deaf, even if the quirk gets removed from another Eigenstasium OD. 
I'd imagine the same happens even if you start with the deaf quirk on your character and get rid of it via OD.

**Comment by ArcaneDefence** [comment]
Sounds like a lot more of a problem with eigenstasium applying quirks which aren't meant to be removable

**Comment by Ghilker** [comment]
the title is a bit misleading, the issue is with eigenstasium giving a quirk than with the quirk itself

**Comment by Onule** [comment]
> 
> 
> the title is a bit misleading, the issue is with eigenstasium giving a quirk than with the quirk itself

The last stage of eigenstasium's OD effects rerolls your character's looks and quirks, it's intentional.
I've only had issues with the Deaf quirk when I OD'd on it twice, inacusiate, upgraded cyber ears, admin aheal did not fix it.

**Comment by timothymtorres** [comment]
> Sounds like a lot more of a problem with eigenstasium applying quirks which aren't meant to be removable

I agree with this.  Some quirks are linked to the mind and some are linked to the body.  Removing a physical quirk like blindness or paraplegic would not magically heal the mob due to these being roundstart quirks.  If the player polymorphed into a new mob they will still have the quirk but lose the physical deformity.

This does bring up an interesting point:  If quirks can be activated midround and cause physical ailments to the mob, then perhaps this should be disabled.  Should quirks be able to be applied or removed during a round?
